### DISCUSSION TITLE
Possibility to access data from one store to another for accessors & mutators

### DISCUSSION THREAD

**Discussion by whitespacecode** [CodeDredd/pinia-orm#341] [Q&A] [answered]
In pinia you have the possibility to access other stores to use getters and setters [Link](https://pinia.vuejs.org/core-concepts/getters.html#accessing-other-stores-getters)
Would it be possible in pinia-orm to use another (related) model's data to use in an accessor?
Or do i need to setup a custom repository [Link](https://pinia-orm.codedredd.de/guide/repository/getting-started#abstract-custom-repository)
 
//Parent
```
import { Model } from "pinia-orm";
import { Person } from "./Person";

export class Adult extends Model {
    static entity = "adults";

    static fields() {
        return {
            id: this.attr(""),
            person_id: this.attr(""),
            person: this.belongsTo(Person, "person_id"),
            extra_data: this.attr(""),
        };
    }
}
```
//Child
```
import { Model } from "pinia-orm";
import { Adult } from "./Adult";

export class Person extends Model {
  static entity = 'person';

  static fields () {
       return {
           id: this.attr(null),
           name: this.attr('')
       }
   }

   get checkForX () {
        return this.name.toLowerCase().includes("x") || Adult.extra_data.includes("x");
    }
}
```

**Comment by CodeDredd** [comment, marked as answer]
With custom repositories you can define what you like.
I think it should be possible to access related data with an getter. But i think the solution i post you is not very performant on big data.
The way you posted child doesn't make sense to access parent data. You need some relation inside the model and not importing the model.

````ts
import { Model } from "pinia-orm";
import { Adult } from "./Adult";

export class Person extends Model {
  static entity = 'person';

  static fields () {
       return {
           id: this.attr(null),
           name: this.attr('')
           adults: this.hasMany(Adult, 'person_id')
       }
   }

   get checkForX () {
        return this.name.toLowerCase().includes("x") || this.adults.map(adult => adult.extra_data).includes("x");
    }
}
````

Haven't tested the code above if it works. But i think you also need to make sure to eagerload the relation if you use this getter.
